src/main/java/org/datadog/jmxfetch/JMXAttribute.java

```diff
@@ -351,6 +355,89 @@ MBeanAttributeInfo getAttribute() {
         return attribute;
     }
 
+    public ObjectName getBeanName() {
+      return beanName;
+    }
+
+    /**
+     * Get attribute alias.
+     *
+     * In order, tries to:
+     * * Use `alias_match` to generate an alias with a regular expression
+     * * Use `alias` directly
+     * * Create an generic alias prefixed with user's `metric_prefix` preference or default to `jmx`
+     *
+     * Argument(s):
+     * * (Optional) `field`
+     *   `Null` for `JMXSimpleAttribute`.
+     */
+    protected String getAlias(String field) {
+        String alias = null;
+
+        Filter include = getMatchingConf().getInclude();
+        LinkedHashMap<String, Object> conf = getMatchingConf().getConf();
+
+        String fullAttributeName =(field!=null)?(getAttribute().getName() + "." + field):(getAttribute().getName());
+
+        if (include.getAttribute() instanceof LinkedHashMap<?, ?>) {
+            LinkedHashMap<String, LinkedHashMap<String, String>> attribute = (LinkedHashMap<String, LinkedHashMap<String, String>>) (include.getAttribute());
+            if (attribute.get(fullAttributeName).get("alias_match") != null) {
+              Pattern p = Pattern.compile(attribute.get(fullAttributeName).get("alias_match"));
+              Matcher m = p.matcher(getBeanName().getCanonicalName() + "." + fullAttributeName);
```

Comment:
@alz I am wondering if we should not match on the bean names only, i.e. drop the the attribute name. What do you think ?